src/Gateway/TransactionGateway.php

```diff
@@ -66,4 +66,16 @@ public function sendPending(int $ct, $lang = 'en')
         $server_output = $this->checkResponse($response, "Retrieving transaction feed!");
         return json_decode($server_output);
     }
+
+    /**
+     * @throws TwikeyException
+     * @throws ClientExceptionInterface
+     */
+    public function cancel(?string $id, ?string $ref, $lang = 'en')
+    {
+        $response = $this->request('DELETE', sprintf('/creditor/transaction%s%s%s', isset($id) || isset($ref) ? '?' : null, isset($id) ? "id=$id" : null, isset($ref) ? sprintf("%ref=$ref", isset($id) ? '&' : null) : null, [], $lang));
```

Comment:
Hi @rvvincelli , please simplify the derivation of the params. Derive the URL params before the call and avoid such long (if-then-else) one liner. Thx.